src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/Grid.cs

```diff
@@ -285,21 +285,16 @@ public ColumnDefinitionCollection ColumnDefinitions
             }
             set
             {
-                _data ??= new ExtendedData();
-                _data.ColumnDefinitions ??= new ColumnDefinitionCollection(this);
-                _data.ColumnDefinitions.Clear();
-                if (value == null)
+                if (value?.Owner is not null)
```

Comment:
if `value?.Owner == this` you should silently exit

Reply:
Added this check

Reply:
Updated